src/java.base/share/classes/java/nio/channels/FileChannel.java

```diff
@@ -642,7 +642,8 @@ public abstract long transferTo(long position, long count,
      * number of bytes will be transferred if the source channel has fewer than
      * {@code count} bytes remaining, or if the source channel is non-blocking
      * and has fewer than {@code count} bytes immediately available in its
-     * input buffer.
+     * input buffer. No bytes are transferred if the source channel has reached
+     * the end-of-stream in which case zero will be returned.
```

Comment:
Thanks for dropping the apiNote that mentions -1 and return value.

The update is mostly okay, I might re-word it slightly to "No bytes are transferred, and zero is returned, if the source has reached end-of-stream".

Reply:
Thanks Alan, just modified as you suggested, I think it's better.